From 491ce40b7f3127ce6b715219df0f3baa73c71fc0 Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Fri, 15 May 2020 14:54:33 -0700
Subject: [PATCH] MinGW doesn't do signals.

---
 libguile/null-threads.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/libguile/null-threads.h b/libguile/null-threads.h
--- a/libguile/null-threads.h
+++ b/libguile/null-threads.h
@@ -82,10 +82,17 @@ scm_i_sched_yield (void)
 
 /* Signals
  */
+#ifdef __MINGW32__
+typedef unsigned int sigset_t;
+#endif
 static inline int
 scm_i_pthread_sigmask (int how, const sigset_t *set, sigset_t *oldset)
 {
+#ifdef __MINGW32__
+  return 0;
+#else
   return sigprocmask (how, set, oldset);
+#endif
 }
 
 /* Mutexes
 
-- 
2.26.0

