From eb6137334bf60be48fdae68cabdb9609cc4bfb21 Mon Sep 17 00:00:00 2001
From: John Ralls <jralls@ceridwen.us>
Date: Sun, 26 May 2019 16:35:33 -0700
Subject: [PATCH 2/3] MinGW: Don't break build when including libguile.h

---
 libguile/iselect.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libguile/iselect.h b/libguile/iselect.h
index 945ad14af..c2f42d38b 100644
--- a/libguile/iselect.h
+++ b/libguile/iselect.h
@@ -29,7 +29,11 @@
 /* Needed for FD_SET on some systems.  */
 #include <sys/types.h>
 
+#ifdef __MINGW32__
+#include <winsock2.h>
+#else
 #include <sys/select.h>
+#endif
 
 SCM_API int scm_std_select (int fds,
 			    fd_set *rfds,
-- 
2.21.0

