$ diff -u configure.ac~ configure.ac
--- a/configure.ac       2024-03-19 14:49:47.000000000 -0700
+++ b/configure.ac        2024-04-19 15:26:08.337623400 -0700
@@ -238,7 +238,7 @@
 )

 if test "$aqbanking_backends" = "all"; then
-  aqbanking_backends="aqhbci aqofxconnect aqebics aqpaypal aqnone aqgivve"
+  aqbanking_backends="aqhbci aqofxconnect aqebics aqpaypal aqnone"
 fi

 for f in ${aqbanking_backends}; do

