--- a/src/tools/aqbanking-cli/Makefile.am
+++ b/src/tools/aqbanking-cli/Makefile.am
@@ -37,6 +37,9 @@ aqbanking_cli_SOURCES= \
   control.c
 
 aqbanking_cli_LDFLAGS=
+aqhbci_tool4_LDFLAGS=
+aqpaypal_tool_LDFLAGS=
+aqebics_tool_LDFLAGS=
 #-Wl,-static
 
 DEFS += -DLOCALEDIR=\"$(localedir)\" -DAQBANKING_CLI_VERSION_STRING=\"@AQBANKING_CLI_VERSION_STRING@\"
@@ -68,6 +68,10 @@ aqebics_tool_LDADD= \
 if IS_WINDOWS
 
 aqbanking_cli_LDFLAGS+=-mconsole
+aqhbci_tool4_LDFLAGS+=-mconsole
+aqpaypal_tool_LDFLAGS+=-mconsole
+aqebics_tool_LDFLAGS+=-mconsole
+
 
 endif
 

