--- a/tools/typemaker2/Makefile.in	2021-11-13 10:12:21.000000000 -0800
+++ b/tools/typemaker2/Makefile.in	2021-11-24 13:31:46.720542300 -0800
@@ -125,8 +125,9 @@
 am__DEPENDENCIES_1 =
 @IS_WINDOWS_TRUE@am__DEPENDENCIES_2 = $(am__DEPENDENCIES_1)
 typemaker2_DEPENDENCIES =  \
+	c/libtm2_c.la \
 	$(top_builddir)/src/$(gwenhywfar_internal_libname) \
-	c/libtm2_c.la $(am__DEPENDENCIES_2)
+	$(am__DEPENDENCIES_2)
 AM_V_lt = $(am__v_lt_@AM_V@)
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
 am__v_lt_0 = --silent
@@ -560,8 +561,10 @@
 
 
 #  tm_enum.c
-typemaker2_LDADD = $(top_builddir)/src/$(gwenhywfar_internal_libname) \
-	c/libtm2_c.la $(am__append_1)
+typemaker2_LDADD = \
+	c/libtm2_c.la \
+	$(top_builddir)/src/$(gwenhywfar_internal_libname) \
+	$(am__append_1)
 @IS_WINDOWS_TRUE@typemaker2_LDFLAGS = -mconsole
 EXTRA_DIST = $(typefiles)
 all: all-recursive
--- a/tools/typemaker2/c/Makefile.in	2021-11-13 10:12:21.000000000 -0800
+++ b/tools/typemaker2/c/Makefile.in	2021-11-24 13:39:45.614600400 -0800
@@ -456,7 +456,7 @@
 with_gui_gtk2 = @with_gui_gtk2@
 with_gui_gtk3 = @with_gui_gtk3@
 with_gui_qt4 = @with_gui_qt4@
-INCLUDES = @all_includes@ -I$(top_builddir) -I../ -I../../ -I$(top_srcdir)/tools -I$(gwenhywfar_symlinkdir)
+INCLUDES = @all_includes@ -I$(top_builddir) -I${srcdir}/../ -I${srcdir}/../../ -I$(top_srcdir)/tools -I$(gwenhywfar_symlinkdir)
 noinst_LTLIBRARIES = libtm2_c.la
 libtm2_c_la_SOURCES = \
  tm_builder_c.c \
