--- a/Makefile.in	2019-09-20 11:59:23.000000000 -0700
+++ b/Makefile.in	2019-09-22 12:31:05.662210400 -0700
@@ -1212,7 +1212,7 @@
 clean-local: clean-local-check
 .PHONY: clean-local
 clean-local-check:
-	rm -rf apidoc gwenhywfar5
+	rm -rf apidoc
 
 listdoc.h: $(top_builddir)/admin/mklistdoc
 	admin/mklistdoc -v -I $(top_srcdir)/src/base `find "$(top_builddir)/gwenhywfar5/gwenhywfar" -name "*.h" | LC_ALL=C sort` >listdoc.h
