diff --git a/jhbuild/modtypes/__init__.py b/jhbuild/modtypes/__init__.py
index 2919a9f9..8b5f915f 100644
--- a/jhbuild/modtypes/__init__.py
+++ b/jhbuild/modtypes/__init__.py
@@ -314,13 +314,15 @@ them into the prefix."""
         save_broken_tree = False
         broken_name = destdir + '-broken'
         destdir_prefix = os.path.join(destdir, stripped_prefix)
-        new_contents = fileutils.accumulate_dirtree_contents(destdir_prefix)
+        new_contents = []
         errors = []
+        num_copied = 0
         if os.path.isdir(destdir_prefix):
             logging.info(_('Moving temporary DESTDIR %r into build prefix') % (destdir, ))
             num_copied = self._process_install_files(destdir, destdir_prefix,
                                                      buildscript.config.prefix,
                                                      errors)
+            new_contents = fileutils.accumulate_dirtree_contents(destdir_prefix)
 
             # Now the destdir should have a series of empty directories:
             # $JHBUILD_PREFIX/_jhbuild/root-foo/$JHBUILD_PREFIX
@@ -345,7 +347,8 @@ them into the prefix."""
             # Even if there are some files outside the DESTDIR, count that as success for now; we just warn
             install_succeeded = True
         else:
-            save_broken_tree = True
+            install_succeeded = True
+#            save_broken_tree = True
 
         if save_broken_tree:
             if os.path.exists(broken_name):
diff --git a/jhbuild/modtypes/cmake.py b/jhbuild/modtypes/cmake.py
index 1bd44e32..4064faaa 100644
--- a/jhbuild/modtypes/cmake.py
+++ b/jhbuild/modtypes/cmake.py
@@ -143,9 +143,9 @@ class CMakeModule(MakeModule, NinjaModule, DownloadableModule):
         buildscript.set_action(_('Installing'), self)
         destdir = self.prepare_installroot(buildscript)
         if self.use_ninja:
-            self.ninja(buildscript, 'install', env={'DESTDIR': destdir})
+            self.ninja(buildscript, 'install')
         else:
-            self.make(buildscript, 'install DESTDIR={}'.format(destdir))
+            self.make(buildscript, 'install')
         self.process_install(buildscript, self.get_revision())
     do_install.depends = [PHASE_BUILD]
 
diff --git a/jhbuild/utils/systeminstall.py b/jhbuild/utils/systeminstall.py
index 610be80d..50c56aab 100644
--- a/jhbuild/utils/systeminstall.py
+++ b/jhbuild/utils/systeminstall.py
@@ -45,7 +45,7 @@ def get_installed_pkgconfigs(config):
     pkgs = []
     for line in stdout.splitlines():
         pkg, rest = line.split(None, 1)
-        pkgs.append(pkg)
+        pkgs.append(pkg.lstrip('/'))

     # see if we can get the versions "the easy way"
     try:
