--- /usr/lib/python2.7/site-packages/jhbuild/modtypes/cmake.py	2017-06-01 12:26:10.000000000 -0700
+++ /usr/lib/python2.7/site-packages/jhbuild/modtypes/cmake.py	2017-08-31 10:46:42.719250200 -0700
@@ -159,11 +159,9 @@
         if self.use_ninja:
             self.ensure_ninja_binary()
             extra_env = self.extra_env or {}
-            extra_env['DESTDIR'] = destdir
             buildscript.execute(self.ninja_binary + ' install', cwd=builddir, extra_env=extra_env)
         else:
-            self.make(buildscript, 'install DESTDIR={}'.format(destdir))
-        self.process_install(buildscript, self.get_revision())
+            self.make(buildscript, 'install')
     do_install.depends = [PHASE_BUILD]
 
     def xml_tag_and_attrs(self):
--- /usr/lib/python2.7/site-packages/jhbuild/modtypes/autotools.py	2017-06-01 12:26:10.000000000 -0700
+++ /usr/lib/python2.7/site-packages/jhbuild/modtypes/autotools.py	2017-08-31 11:00:39.148673100 -0700
@@ -309,9 +309,7 @@
                 packagedb.uninstall(self.name)
 
         buildscript.set_action(_('Installing'), self)
-        destdir = self.prepare_installroot(buildscript)
-        self.make(buildscript, self.makeinstallargs or 'install', makeargs='DESTDIR={}'.format(destdir))
-        self.process_install(buildscript, self.get_revision())
+        self.make(buildscript, self.makeinstallargs or 'install')
 
     do_install.depends = [PHASE_BUILD]
     do_install.error_phases = [PHASE_CONFIGURE]
