diff --git a/docs/api/Makefile.am b/docs/api/Makefile.am
index a1ffd3a7..e88ea929 100644
--- a/docs/api/Makefile.am
+++ b/docs/api/Makefile.am
@@ -268,9 +268,8 @@ clean-local:
 	-rm -rf $(SOURCE_CODE_DIR) $(EXAMPLES_DIR) $(SCAN_DIR)/*.txt $(SGML_DIR) $(GTKDOC_MKDB_OUTPUT_DIR)
 	-rm -f *.stamp *.types index.sgml
 	@( if [ z"$(builddir)" != z"$(srcdir)" ] ; then \
-    	chmod -R u+w $(builddir)/src && rm -rf $(builddir)/src ; \
-    	chmod -R u+w $(builddir)/images && rm -rf $(builddir)/images ; \
-    	chmod -R u+w $(builddir)/*.png && rm -rf $(builddir)/*.png ; \
+	find $(builddir) -name 'src' -o -name 'images' -o -name '*.png' -exec chmod u+w '{}' ';' ; \
+	rm -rf $(builddir)/src $(builddir)/images $(builddir)/*.png ; \
 	 fi ; )
 
 distclean-local: clean-local
