--- a/Tools/WebKitTestRunner/GNUmakefile.am	
+++ a/Tools/WebKitTestRunner/GNUmakefile.am	
@@ -155,7 +155,10 @@ code_generation_dependencies = \
 	Source/WebCore/bindings/scripts/InFilesParser.pm \
 	Source/WebCore/bindings/scripts/generate-bindings.pl
 
-DerivedSources/InjectedBundle/JS%.cpp: Tools/WebKitTestRunner/InjectedBundle/Bindings/%.idl $(code_generation_dependencies)
+$(top_builddir)/DerivedSources/InjectedBundle:
+	$(AM_V_at)mkdir -p $@
+
+DerivedSources/InjectedBundle/JS%.cpp: Tools/WebKitTestRunner/InjectedBundle/Bindings/%.idl $(code_generation_dependencies) $(top_builddir)/DerivedSources/InjectedBundle
 	$(AM_V_GEN)$(PERL) \
 		-I $(srcdir)/Source/WebCore/bindings/scripts \
 		-I $(srcdir)/Tools/WebKitTestRunner/InjectedBundle/Bindings \
--