diff --git a/glib/gtimezone.c b/glib/gtimezone.c
index b61529f..2e2369c 100644
--- a/glib/gtimezone.c
+++ b/glib/gtimezone.c
@@ -893,7 +893,7 @@ init_zone_from_rules (GTimeZone    *gtz,
           ++info_index;
           skip_first_std_trans = TRUE;
          }
-      else if (rules[ri].std_offset || rules[ri].dlt_offset)
+      else
         {
           const guint start_year = rules[ri].start_year;
           const guint end_year = rules[ri + 1].start_year;
