--- a/src/sar/sar.c	Wed Dec 31 05:29:02 2014
+++ b/src/sar/sar.c	Sun Mar 29 18:40:06 2015
@@ -614,7 +614,8 @@
 
 
 int GWEN_Sar_AddAndDigestFileLink(GWEN_SAR *sr, const GWEN_SAR_FILEHEADER *fh, GWEN_MDIGEST *md) {
-#if _BSD_SOURCE || _XOPEN_SOURCE >= 500 || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) || _POSIX_C_SOURCE >= 200112L || defined(OS_DARWIN)
+#if ((_BSD_SOURCE || _XOPEN_SOURCE >= 500 || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) || _POSIX_C_SOURCE >= 200112L) && !defined(__MINGW32__)) || defined(OS_DARWIN)
+
   int rv;
   const char *fname;
   GWEN_BUFFER *hbuf;
@@ -1352,7 +1353,7 @@
 
 
 int GWEN_Sar_ExtractAndDigestFileLink(GWEN_SAR *sr, const GWEN_SAR_FILEHEADER *fh, int checkOnly) {
-#if _BSD_SOURCE || _XOPEN_SOURCE >= 500 || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) || _POSIX_C_SOURCE >= 200112L || defined(OS_DARWIN)
+#if ((_BSD_SOURCE || _XOPEN_SOURCE >= 500 || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) || _POSIX_C_SOURCE >= 200112L) && !defined(__MINGW32__)) || defined(OS_DARWIN)
   int rv;
   const char *fname;
   uint32_t perms;

--- a/gui/testdialogs/dlg_test.c	Sun Oct 12 06:12:22 2014
+++ b/gui/testdialogs/dlg_test.c	Sun Mar 29 19:11:48 2015
@@ -18,11 +18,6 @@
 #include <gwenhywfar/pathmanager.h>
 
 
-#ifdef OS_WIN32
-# include <windows.h>
-# define sleep(x) Sleep(x)
-#endif
-
 #include <unistd.h>
 
 
--- a/gui/testdialogs/dlg_test2.c	Sun Oct 12 06:12:22 2014
+++ b/gui/testdialogs/dlg_test2.c	Sun Mar 29 19:11:48 2015
@@ -16,11 +16,6 @@
 #include <gwenhywfar/pathmanager.h>
 
 
-#ifdef OS_WIN32
-# include <windows.h>
-# define sleep(x) Sleep(x)
-#endif
-
 #include <unistd.h>
 
 
--- a/gui/gtk2/libtest.c	Sun Oct 12 06:12:22 2014
+++ b/gui/gtk2/libtest.c	Mon Mar 30 06:32:34 2015
@@ -19,12 +19,6 @@
 #include <gwenhywfar/dialog.h>
 #include <gwenhywfar/debug.h>
 
-
-#ifdef OS_WIN32
-# include <windows.h>
-# define sleep(x) Sleep(x)
-#endif
-
 #include <unistd.h>
 
 
