--- a/src/sar/sar.c	Wed Dec 31 05:29:02 2014
+++ b/src/sar/sar.c	Sun Mar 29 18:40:06 2015
@@ -614,7 +614,8 @@
 
 
 int GWEN_Sar_AddAndDigestFileLink(GWEN_SAR *sr, const GWEN_SAR_FILEHEADER *fh, GWEN_MDIGEST *md) {
-#if _BSD_SOURCE || _XOPEN_SOURCE >= 500 || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) || _POSIX_C_SOURCE >= 200112L || defined(OS_DARWIN)
+#if ((_BSD_SOURCE || _XOPEN_SOURCE >= 500 || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) || _POSIX_C_SOURCE >= 200112L) && !defined(__MINGW32__)) || defined(OS_DARWIN)
+
   int rv;
   const char *fname;
   GWEN_BUFFER *hbuf;
@@ -1352,7 +1353,7 @@
 
 
 int GWEN_Sar_ExtractAndDigestFileLink(GWEN_SAR *sr, const GWEN_SAR_FILEHEADER *fh, int checkOnly) {
-#if _BSD_SOURCE || _XOPEN_SOURCE >= 500 || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) || _POSIX_C_SOURCE >= 200112L || defined(OS_DARWIN)
+#if ((_BSD_SOURCE || _XOPEN_SOURCE >= 500 || (_XOPEN_SOURCE && _XOPEN_SOURCE_EXTENDED) || _POSIX_C_SOURCE >= 200112L) && !defined(__MINGW32__)) || defined(OS_DARWIN)
   int rv;
   const char *fname;
   uint32_t perms;

--- a/gui/testdialogs/dlg_test.c	Sun Oct 12 06:12:22 2014
+++ b/gui/testdialogs/dlg_test.c	Sun Mar 29 19:11:48 2015
@@ -18,11 +18,6 @@
 #include <gwenhywfar/pathmanager.h>
 
 
-#ifdef OS_WIN32
-# include <windows.h>
-# define sleep(x) Sleep(x)
-#endif
-
 #include <unistd.h>
 
 
--- a/gui/testdialogs/dlg_test2.c	Sun Oct 12 06:12:22 2014
+++ b/gui/testdialogs/dlg_test2.c	Sun Mar 29 19:11:48 2015
@@ -16,11 +16,6 @@
 #include <gwenhywfar/pathmanager.h>
 
 
-#ifdef OS_WIN32
-# include <windows.h>
-# define sleep(x) Sleep(x)
-#endif
-
 #include <unistd.h>
 
 
--- a/gui/gtk2/libtest.c	Sun Oct 12 06:12:22 2014
+++ b/gui/gtk2/libtest.c	Mon Mar 30 06:32:34 2015
@@ -19,12 +19,6 @@
 #include <gwenhywfar/dialog.h>
 #include <gwenhywfar/debug.h>
 
-
-#ifdef OS_WIN32
-# include <windows.h>
-# define sleep(x) Sleep(x)
-#endif
-
 #include <unistd.h>
 
 
--- a/src/crypt3/cryptkey_be.h	2014-10-12 15:12:22 +0200
+++ b/src/crypt3/cryptkey_be.h	2015-05-07 20:51:40 +0200
@@ -18,22 +18,22 @@
 
 
 
-typedef int (*GWEN_CRYPT_KEY_SIGN_FN)(GWEN_CRYPT_KEY *k,
+typedef GWENHYWFAR_CB int (*GWEN_CRYPT_KEY_SIGN_FN)(GWEN_CRYPT_KEY *k,
                                       const uint8_t *pInData,
                                       uint32_t inLen,
                                       uint8_t *pSignatureData,
                                       uint32_t *pSignatureLen);
-typedef int (*GWEN_CRYPT_KEY_VERIFY_FN)(GWEN_CRYPT_KEY *k,
+typedef GWENHYWFAR_CB int (*GWEN_CRYPT_KEY_VERIFY_FN)(GWEN_CRYPT_KEY *k,
                                         const uint8_t *pInData,
                                         uint32_t inLen,
                                         const uint8_t *pSignatureData,
                                         uint32_t signatureLen);
-typedef int (*GWEN_CRYPT_KEY_ENCIPHER_FN)(GWEN_CRYPT_KEY *k,
+typedef GWENHYWFAR_CB int (*GWEN_CRYPT_KEY_ENCIPHER_FN)(GWEN_CRYPT_KEY *k,
     const uint8_t *pInData,
     uint32_t inLen,
     uint8_t *pOutData,
     uint32_t *pOutLen);
-typedef int (*GWEN_CRYPT_KEY_DECIPHER_FN)(GWEN_CRYPT_KEY *k,
+typedef GWENHYWFAR_CB int (*GWEN_CRYPT_KEY_DECIPHER_FN)(GWEN_CRYPT_KEY *k,
     const uint8_t *pInData,
     uint32_t inLen,
     uint8_t *pOutData,
--- a/src/crypt3/cryptkeysym.c	2014-10-12 15:12:22 +0200
+++ b/src/crypt3/cryptkeysym.c	2015-05-07 20:51:04 +0200
@@ -29,7 +29,7 @@
 
 
 
-int GWEN_Crypt_KeySym_Encipher(GWEN_CRYPT_KEY *k,
+GWENHYWFAR_CB int GWEN_Crypt_KeySym_Encipher(GWEN_CRYPT_KEY *k,
                                const uint8_t *pInData,
                                uint32_t inLen,
                                uint8_t *pOutData,
@@ -53,7 +53,7 @@
 
 
 
-int GWEN_Crypt_KeySym_Decipher(GWEN_CRYPT_KEY *k,
+GWENHYWFAR_CB int GWEN_Crypt_KeySym_Decipher(GWEN_CRYPT_KEY *k,
                                const uint8_t *pInData,
                                uint32_t inLen,
                                uint8_t *pOutData,
--- a/src/crypt3/cryptkeysym_p.h	2014-10-12 15:12:22 +0200
+++ b/src/crypt3/cryptkeysym_p.h	2015-05-07 20:50:41 +0200
@@ -29,12 +29,12 @@
 
 static GWENHYWFAR_CB void GWEN_Crypt_KeySym_freeData(void *bp, void *p);
 
-static int GWEN_Crypt_KeySym_Encipher(GWEN_CRYPT_KEY *k,
+static GWENHYWFAR_CB int GWEN_Crypt_KeySym_Encipher(GWEN_CRYPT_KEY *k,
                                       const uint8_t *pInData,
                                       uint32_t inLen,
                                       uint8_t *pOutData,
                                       uint32_t *pOutLen);
-static int GWEN_Crypt_KeySym_Decipher(GWEN_CRYPT_KEY *k,
+static GWENHYWFAR_CB int GWEN_Crypt_KeySym_Decipher(GWEN_CRYPT_KEY *k,
                                       const uint8_t *pInData,
                                       uint32_t inLen,
                                       uint8_t *pOutData,
