--- a/configure.ac	2016-04-09 23:48:38.000000000 -0700
+++ b/configure.ac	2021-09-29 14:11:36.180373200 -0700
@@ -36,6 +36,8 @@
 AM_MAINTAINER_MODE([enable])
 m4_include([Source/autotools/SetupAutomake.m4])
 
+LT_INIT
+GTK_DOC_CHECK(1.0)
 ######################################################################################
 # Processing of configuration files
 ######################################################################################
