--- webkitgtk-2.4.11/Source/JavaScriptCore/runtime/ArgumentsIteratorConstructor.cpp	2015-09-03 00:12:57.000000000 +0300
+++ webkitgtk-2.4.11/Source/JavaScriptCore/runtime/ArgumentsIteratorConstructor.cpp	2015-09-03 00:12:57.000000000 +0300
@@ -31,6 +31,7 @@
 #include "JSCJSValueInlines.h"
 #include "JSCellInlines.h"
 #include "JSGlobalObject.h"
+#include "StructureInlines.h"
 
 namespace JSC {
 
--- webkitgtk-2.4.11/Source/JavaScriptCore/runtime/JSPromiseFunctions.cpp	2015-09-03 00:12:58.000000000 +0300
+++ webkitgtk-2.4.11/Source/JavaScriptCore/runtime/JSPromiseFunctions.cpp	2015-09-03 00:12:58.000000000 +0300
@@ -35,6 +35,7 @@
 #include "JSPromiseConstructor.h"
 #include "JSPromiseDeferred.h"
 #include "NumberObject.h"
+#include "StructureInlines.h"
 
 namespace JSC {
 
--- webkitgtk-2.4.11/Source/JavaScriptCore/runtime/MapConstructor.cpp	2015-09-03 00:13:00.000000000 +0300
+++ webkitgtk-2.4.11/Source/JavaScriptCore/runtime/MapConstructor.cpp	2015-09-03 00:13:00.000000000 +0300
@@ -32,6 +32,7 @@
 #include "JSGlobalObject.h"
 #include "JSMap.h"
 #include "MapPrototype.h"
+#include "StructureInlines.h"
 
 namespace JSC {
 
--- webkitgtk-2.4.11/Source/JavaScriptCore/runtime/MapIteratorConstructor.cpp	2015-09-03 00:13:01.000000000 +0300
+++ webkitgtk-2.4.11/Source/JavaScriptCore/runtime/MapIteratorConstructor.cpp	2015-09-03 00:13:01.000000000 +0300
@@ -31,6 +31,7 @@
 #include "JSGlobalObject.h"
 #include "JSMapIterator.h"
 #include "MapIteratorPrototype.h"
+#include "StructureInlines.h"
 
 namespace JSC {
 
--- webkitgtk-2.4.11/Source/JavaScriptCore/runtime/SetConstructor.cpp	2015-09-03 00:13:02.000000000 +0300
+++ webkitgtk-2.4.11/Source/JavaScriptCore/runtime/SetConstructor.cpp	2015-09-03 00:13:02.000000000 +0300
@@ -33,6 +33,7 @@
 #include "JSSet.h"
 #include "MapData.h"
 #include "SetPrototype.h"
+#include "StructureInlines.h"
 
 namespace JSC {
 
--- webkitgtk-2.4.11/Source/JavaScriptCore/runtime/SetIteratorConstructor.cpp	2015-09-03 00:13:04.000000000 +0300
+++ webkitgtk-2.4.11/Source/JavaScriptCore/runtime/SetIteratorConstructor.cpp	2015-09-03 00:13:04.000000000 +0300
@@ -31,6 +31,7 @@
 #include "JSGlobalObject.h"
 #include "JSSetIterator.h"
 #include "SetIteratorPrototype.h"
+#include "StructureInlines.h"
 
 namespace JSC {
 
--- webkitgtk-2.4.11/Source/JavaScriptCore/runtime/WeakMapConstructor.cpp	2015-09-03 00:13:06.000000000 +0300
+++ webkitgtk-2.4.11/Source/JavaScriptCore/runtime/WeakMapConstructor.cpp	2015-09-03 00:13:06.000000000 +0300
@@ -30,6 +30,7 @@
 #include "JSCellInlines.h"
 #include "JSGlobalObject.h"
 #include "JSWeakMap.h"
+#include "StructureInlines.h"
 #include "WeakMapPrototype.h"
 
 namespace JSC {
