--- a/src/tools/cardcommander/cardcommander.cpp	2021-09-11 07:09:12.000000000 -0700
+++ b/src/tools/cardcommander/cardcommander.cpp	2025-09-14 13:24:29.390745100 -0700
@@ -45,7 +45,7 @@
 using namespace std;
 
 #define k_PRG_VERSION_INFO \
-  "cardcommander v0.4  (part of libchipcard v"k_CHIPCARD_VERSION_STRING")\n"\
+  "cardcommander v0.4  (part of libchipcard v" k_CHIPCARD_VERSION_STRING ")\n"\
   "(c) 2006 Martin Preuss<martin@libchipcard.de>\n" \
   "This program is free software licensed under GPL.\n"\
   "See COPYING for details.\n"
